provider/pkg/provider/remote/connection.go

```diff
@@ -61,16 +66,17 @@ func (c *Connection) Annotate(a infer.Annotator) {
 	a.Describe(&c.PrivateKeyPassword, "The password to use in case the private key is encrypted.")
 	a.Describe(&c.AgentSocketPath, "SSH Agent socket path. Default to environment variable SSH_AUTH_SOCK if present.")
 	a.Describe(&c.DialErrorLimit, "Max allowed errors on trying to dial the remote host. -1 set count to unlimited. Default value is 10.")
+	a.Describe(&c.Proxy, "The connection settings for the bastion/proxy host.")
 	a.SetDefault(&c.DialErrorLimit, dialErrorDefault)
 	a.Describe(&c.PerDialTimeout, "Max number of seconds for each dial attempt. 0 implies no maximum. Default value is 15 seconds.")
 	a.SetDefault(&c.PerDialTimeout, 15)
 }
 
-func (con *Connection) SShConfig() (*ssh.ClientConfig, error) {
+func (con *connectionBase) SShConfig() (*ssh.ClientConfig, error) {
 	config := &ssh.ClientConfig{
 		User:            *con.User,
 		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
-		Timeout:         time.Second * time.Duration(con.PerDialTimeout),
+		Timeout:         time.Second * time.Duration(*con.PerDialTimeout),
```

Comment:
con.PerDialTimeout never nil presumably?

Reply:
Nope: https://github.com/pulumi/pulumi-command/blob/be77edab46000128798caaf5b67a9673558e75e4/provider/pkg/provider/remote/connection.go#L72

If the user didn't supply it, we would have added the default value.